Separate non-res channels into separate field in G3File

jlashner opened the pull request:
As we discussed today, the easiest way at the moment to resolve the discrepancy between the det-db and non-resonator channels existing in the data is to separate them from the 'signal' array in the bookbinder. This PR selects channels that don't have an associated stream-id or tunefile and puts them into a separate array instead of adding them to `signal`.

The implementation here may need to change if we change the readout_ids of fixed tones so they aren't marked with NONE.

kmharrington commented:
Question before reviewing, have you run this on books with and without fixed tones?

jlashner commented on sotodlib/io/bookbinder.py:
```diff
@@ -515,7 +518,8 @@ def bind(self, outdir, times, frame_idxs, file_idxs, pbar=False, ancil=None,
             ts = core.G3VectorTime(ts * core.G3Units.s)
-            oframe['signal'] = so3g.G3SuperTimestream(self.readout_ids, ts, data)
+            oframe['signal'] = so3g.G3SuperTimestream(self.readout_ids[res_chans], ts, data[res_chans, :])
+            oframe['non_res_signal'] = so3g.G3SuperTimestream(self.readout_ids[~res_chans], ts, data[~res_chans, :])
```
I am not opposed to changing the name if we figure out a better one, but I think "tones" or even "fixed_tones" is misleading, since it is not exclusively fixed tones. Just anything thats not in the tunefile.

kmharrington replied:
"untracked"?

jlashner replied:
ooo ya that works

kmharrington commented:
Oh also, I notice the python3.10 test actually failed on the changed lines of code.

jlashner commented:
Confirmed that binding observations with no "non-res" channels works, the G3Frame just looks like this:
```
In [13]: print(frame)
Frame (Scan) [
"ancil" (spt3g.core.G3TimesampleMap) => <co-sampled vectors with 50000 samples>{}
"book_id" (spt3g.core.G3String) => "obs_1697368924_satp1_1111110"
"flag_smurfgaps" (spt3g.core.G3VectorBool) => 50000 elements
"frame_num" (spt3g.core.G3Int) => 2
"non_res_signal" (so3g.G3SuperTimestream) => G3SuperTimestream(0, 50000)
"primary" (so3g.G3SuperTimestream) => G3SuperTimestream(9, 50000)
"sample_range" (spt3g.core.G3VectorInt) => [0, 50000]
"signal" (so3g.G3SuperTimestream) => G3SuperTimestream(812, 50000)
"stream_id" (spt3g.core.G3String) => "ufm_mv18"
"tes_biases" (so3g.G3SuperTimestream) => G3SuperTimestream(16, 50000)
]
```
I'll work on fixing the tests.

mhasself reviewed (changes requested):
Still working on this?  I support "untracked".

mhasself commented:
I'm converting this to draft to get it off my radar.  Bring it back to life when you're ready.

jlashner commented:
There is currently a PR to sodetlib in the works that will allow more consistent usage of fixed tones, so I think its important that we merge this in relatively soon.

I've changed the name of the fixed-tone signal to "untracked" as is suggested above and think I fixed the tests.
